### PR TITLE
Clarify tts:fontSize semantics

### LINKED ISSUE #206: Use of 'em' units in tts:fontSize on region element is not well defined.
§8.2.9 needs to place a constraint on the use of 'em' units when expressing tts:fontSize on a region element; alternatively, it could specify that 1em is equal to the initial value of tts:fontSize, namely, 1c, for the purpose of resolving an 'em' unit based font size

### LINKED ISSUE #214: Clarify the semantics of tts:fontSize="2em".
XSL, which TTML uses as a reference for 'em', states that "a relative length is a unit-based value that is measured against the current value of the font-size property" and "when the "font-size" property is
evaluated, the current font-size for use in evaluation is the font-size of the parent element"

While TTML does not specify a definition for the _current_ value of tts:fontSize, using the computed value of tts:fontSize of the parent would be consistent with XSL and the definition of the percentage metric in TTML.

See also #206

### LINKED ISSUE #215: Two value percentage fontSize not fully defined.
See w3c/ttml2#200

### PR DESCRIPTION
**Pull request by palemieux**
Clarify tts:fontSize specified in 'em' units (Closes #214 and closes #206)
Clarify tts:fontSize when specified using two % values (Closes #215)

This PR is intended to replace PR #279

The PR is based on the prose found in TTML2

### CODE REVIEW

**Comment by palemieux** [comment]
See specification at https://rawgit.com/w3c/ttml1/issue-215-214-279-fontsize-semantics-build/index.html

**Review by andreastai — APPROVED** [review]
Looks good to me. Thank you! The examples in the informative notes make it very clear.

File: spec/ttml1.xml
Comment (nigelmegitt) [inline spec/ttml1.xml]: Editorial point: the word "discussion" seems inappropriate for normative requirements. Suggest changing to "details" or even omitting altogether in favour of simple "see below".

File: spec/ttml1.xml
Comment (nigelmegitt) [inline spec/ttml1.xml:4354]: Note for posterity: the rationale for this is described in https://github.com/w3c/ttml2/issues/200#issuecomment-305547161

File: spec/ttml1.xml
Comment (nigelmegitt) [inline spec/ttml1.xml]: Same comment as https://github.com/w3c/ttml2/pull/336/files#r157897318 :

This loses the sense that the font-defined glyph generally needs scaling to the precise display dimensions of the em square, and specifically always needs scaling anamorphically to achieve a different width to the height. I think this needs to be brought back in somehow, though I like the idea of specifying the resultant size of the em square unambiguously.

File: spec/ttml1.xml
Comment (nigelmegitt) [inline spec/ttml1.xml]: I like the idea of defining it in this way. It needs one further concept to be explicitly stated: if a single length value is specified, the resolved value relative to the reference is used for both the width and the height component of the computed value.

This concept is what is intended by the text on line 4290, "this length applies equally to horizontal and vertical ..." but that text is somewhat inadequate in my view: it is unclear that _applies_ means the same as the intended meaning _specifies_. Perhaps I should raise that as a separate issue, but I'd much rather have one pull request that sorts this tortuous font size issue out once and for all.

File: spec/ttml1.xml
Comment (nigelmegitt) [inline spec/ttml1.xml:4355]: See comment below - the word "applies" seems insufficient.

File: spec/ttml1.xml
Comment (nigelmegitt) [inline spec/ttml1.xml:4385]: As in the identical example in w3c/ttml2#336, the text is inconsistent with the span example on line 4339 - it seems odd that paragraph and span elements are not described in the same way. See also https://github.com/w3c/ttml2/pull/336/files#r157896644

File: spec/ttml1.xml
Comment (nigelmegitt) [inline spec/ttml1.xml]: Why drop this note? If I remember correctly it was added after feedback from @palemieux that implementers had difficulty understanding the correct algorithm for calculating the "closes supported value".

File: spec/ttml1.xml
Comment (nigelmegitt) [inline spec/ttml1.xml:4386]: Same comment as https://github.com/w3c/ttml2/pull/336/files#r157899921 :

As per the discussion in https://github.com/w3c/ttml2/issues/200#issuecomment-305547161 the sizes here are in logical pixels - in fact they are dimensionless until multiplied by the size of the pixel unit, whatever that turns out to be. Consider adding a parenthetical _(in logical pixels)_ to avoid confusion with "real" pixels.

File: spec/ttml1.xml
Comment (nigelmegitt) [inline spec/ttml1.xml:4380]: Seems to be missing the useful note as in https://github.com/w3c/ttml2/pull/336/files#diff-5ac39ab39265717d36dfd789d254ce1aR9801 about the use of anamorphic scaling.

**Review by nigelmegitt — CHANGES_REQUESTED** [review]
Generally this is a good change headed in the right direction. Takes a lot of care to review in association with w3c/ttml2#336 - it is probably most efficient use of the group's time to consider both these pull requests together and, when happy with both, merge both. 

I've made some comments on both pull requests.

I've also requested some changes specific to this one, mainly editorial.

File: spec/ttml1.xml
Comment (nigelmegitt) [reply spec/ttml1.xml]: Proposal:

_Note: Presentation processors are expected to scale glyphs so that the height of the em square as defined in the font resource matches the computed font size height, using an appropriate algorithm for the font. If the computed font size width differs from the computed font size height presentation processors additionally need to scale the em square anamorphically by the ratio of the width to the height._

File: spec/ttml1.xml
Comment (skynavga) [reply spec/ttml1.xml]: normally, we say "see prose" which is consistent with XSL-FO and CSS usage

File: spec/ttml1.xml
Comment (skynavga) [inline spec/ttml1.xml]: If we are trying to make this close to TTML2 changes, then this should go back into the table.

**Review by skynavga — CHANGES_REQUESTED** [review]
My overall problem with this PR is that it attempts to incorporate some of the new TTML2 text, but does so only partly, and makes what appear to be arbitrary changes in that text as well. I suggest we focus on (1) identifying what is broken in TTML1 2ed, and (2) fix only that text using the absolutely minimum changes.

File: spec/ttml1.xml
Comment (nigelmegitt) [reply spec/ttml1.xml]: Action on me to file this against TTML2 for discussion there.

File: spec/ttml1.xml
Comment (nigelmegitt) [reply spec/ttml1.xml]: Needs filing against TTML2 also.

File: spec/ttml1.xml
Comment (palemieux) [reply spec/ttml1.xml:4380]: Added.

**Review by nigelmegitt — APPROVED** [review]

File: spec/ttml1.xml
Comment (nigelmegitt) [reply spec/ttml1.xml]: Done as #548.

File: spec/ttml1.xml
Comment (nigelmegitt) [reply spec/ttml1.xml]: Filed as #547

**Comment by palemieux** [comment]
@skynavga Please review the revised PR

File: spec/ttml1.xml
Comment (skynavga) [inline spec/ttml1.xml]: please rewrite to use same structure as TTML2 text; namely, it should read:

if not region element, then relative to nearest ancestor styled element's computed font size; otherwise, relative to the Computed Cell Size

File: spec/ttml1.xml
Comment (skynavga) [inline spec/ttml1.xml:4354]: I wonder if this new paragraph should go here, since "computed value" only appears in the notes farther down and in the paragraph about supported values. It may be better to move this to just prior to the paragraph that reads: "If the relative unit is cell (c), then the referenced value is the Computed Cell Size." However, I can see how putting it here may give guidance to interpreting the 5 paragraphs that precede this. Since I don't have a strong opinion, I can let this stand.

File: spec/ttml1.xml
Comment (skynavga) [inline spec/ttml1.xml]: This note should be placed in the same relative position as in TTML2, i.e., after the paragraph starting with "When a single relative <length> value is specified...".

File: spec/ttml1.xml
Comment (skynavga) [inline spec/ttml1.xml:4372]: Missing the following text present in TTML2:

>If the relative unit is a percentage, then the referenced value is either the computed value of the font size of the nearest styled ancestor element or the computed cell size. If the relative unit is cells (c), then the referenced value is the computed cell size. If the relative unit is EMs (em), then the referenced value is determined as if percentage units were used, where 1em equals 100%.

File: spec/ttml1.xml
Comment (skynavga) [inline spec/ttml1.xml]: This and the following paragraph should match the presentation order in TTML2, i.e., go before the paragraph proscribing negative length values.

File: spec/ttml1.xml
Comment (skynavga) [inline spec/ttml1.xml:4381]: The material added from here to line 4361 should be identical to the TTML2 text, but it is rephrased for no apparent reason.

**Review by skynavga — CHANGES_REQUESTED** [review]
Without completing my detailed review, I have already provided new comments that clearly indicate the unnecessary departure from these changes and the TTML2 text that applies here as well. You need to start a new PR basing it exactly on the TTML2 changes and then propose (and argue for) specific changes that we should adopt both here and in TTML2.

File: spec/ttml1.xml
Comment (palemieux) [reply spec/ttml1.xml]: The note placement is intended to match TTML1 2ED

File: spec/ttml1.xml
Comment (palemieux) [reply spec/ttml1.xml]: Moved to match TTML1 placement

File: spec/ttml1.xml
Comment (palemieux) [reply spec/ttml1.xml]: Modified to match TTML1

File: spec/ttml1.xml
Comment (palemieux) [reply spec/ttml1.xml:4372]: First sentence is already in `tts:fontSize` definition table, and the two following sentences are in fact in the prose.

File: spec/ttml1.xml
Comment (nigelmegitt) [reply spec/ttml1.xml:4381]: This text is in fact present in TTML2. The difference is that TTML2 introduces new text about resolving the c unit horizontally and vertically. I don't think that text is needed here at this time - if it is then please raise a separate issue. We're trying to close this down.

**Review by nigelmegitt — APPROVED** [review]
Approving the new changes - as agreed in today's WG meeting we will merge this since all the substantive comments have been addressed.  Any further issues should be raised separately after the merge.

**Comment by palemieux** [comment]
@skynavga Please review per https://github.com/w3c/ttml1/pull/301#pullrequestreview-91605445 and revised PR

**Comment by nigelmegitt** [comment]
@skynavga ping - last chance to re-review this please. Otherwise I'll have to dismiss your review so we can merge and make progress towards CR for TTML1 Third Ed. It's been 26 days since the WG agreed that all the substantive comments have been addressed - the only reason I'm not going ahead and doing this straight away is because I know you've been working on TTML2 issues recently.

**Comment by skynavga** [comment]
@nigelmegitt give me until tomorrow

File: spec/ttml1.xml
Comment (skynavga) [reply spec/ttml1.xml:4372]: I would prefer that the order of the two notes "Anamorphic ..." and "A glyph's EM square ..." be changed so the latter appears first. This corresponds with the material in the preceding paragraph: EM square is used before referring to anamorphic (two values).

File: spec/ttml1.xml
Comment (skynavga) [inline spec/ttml1.xml]: This paragraph needs to go before the preceding note "In this context, the closest supported value ...", which purpose is to elaborate this paragraph (that now follows the note).

**Review by skynavga — CHANGES_REQUESTED** [review]
I can approve once my two comments on ordering of notes is addressed.

**Comment by palemieux** [comment]
@skynavga Implemented your suggestions. Please review.

**Review by skynavga — APPROVED** [review]
Thanks.

**Pull request merged**
